includes/class-wc-gateway-ppec-cart-handler.php

```diff
@@ -223,11 +299,68 @@ public function enqueue_scripts() {
 					'ajaxurl'     => WC_AJAX::get_endpoint( 'wc_ppec_update_shipping_costs' ),
 				)
 			);
+
+		} elseif ( 'yes' === $settings->use_spb ) {
+			wp_enqueue_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js', array(), null, true );
+			wp_enqueue_script( 'wc-gateway-ppec-smart-payment-buttons', wc_gateway_ppec()->plugin_url . 'assets/js/wc-gateway-ppec-smart-payment-buttons.js', array( 'jquery', 'paypal-checkout-js' ), wc_gateway_ppec()->version, true );
+
+			$data = array(
+				'environment'          => 'sandbox' === $settings->get_environment() ? 'sandbox' : 'production',
+				'locale'               => $settings->get_paypal_locale(),
+				'page'                 => $page,
+				'button_color'         => $settings->button_color,
+				'button_shape'         => $settings->button_shape,
+				'start_checkout_nonce' => wp_create_nonce( '_wc_ppec_start_checkout_nonce' ),
+				'start_checkout_url'   => WC_AJAX::get_endpoint( 'wc_ppec_start_checkout' ),
+			);
+
+			if ( ! is_null(  $page ) ) {
+				if ( 'product' === $page ) {
+					$page_data = array(
+						'button_layout'        => $settings->single_product_button_layout,
+						'button_size'          => $settings->single_product_button_size,
+						'hide_funding_methods' => $settings->single_product_hide_funding_methods,
+						'credit_enabled'       => $settings->single_product_credit_enabled,
+					);
+				} elseif ( 'checkout' === $page && 'yes' === $settings->mark_settings_toggle ) {
+					$page_data = array(
+						'button_layout'        => $settings->mark_button_layout,
+						'button_size'          => $settings->mark_button_size,
+						'hide_funding_methods' => $settings->mark_hide_funding_methods,
+						'credit_enabled'       => $settings->mark_credit_enabled,
+					);
+				} else {
+					$page_data = array(
+						'button_layout'        => $settings->button_layout,
+						'button_size'          => $settings->button_size,
+						'hide_funding_methods' => $settings->hide_funding_methods,
+						'credit_enabled'       => $settings->credit_enabled,
+					);
+				}
+
+				$data = array_merge( $data, $this->convert_to_render_settings( $page_data ) );
```

Comment:
Since these 4 settings are repeated a lot, and are the only ones inspected/modified in `convert_to_render_settings()` - what do you think about combining the retrieval of them with an optional prefix from `$settings` and the "conversion" into a single method? Something like `get_button_settings( $settings, 'single_product' )`?

Reply:
Done in https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/pull/409/commits/86932eefb55ab497b11d51c4714b36a3aa58327d